### DISCUSSION TITLE
Pingdom : Issue with Query-Types

### DISCUSSION THREAD

**Discussion by smm0864** [grafana/grafana-infinity-datasource#115] [Specific APIs] [answered]
I'm trying to query Pingdom API for data, which returns JSON data.
When setting the query type to JSON i'm getting a 'Bad Request'. But when setting it to HTML, i can see in the developer console that the response contains the expected JSON data. 
I tried to debug the code a bit, but couldn't spot a relevant difference between the types. The result parser is a different one. Because of that the returned JSON data can't be evaluated, as the type is HTML.

So my question would be, whats the difference in making requests to the API (in this case pingdom), when changing the query type? I tried also different ways of using the 'Accept' header.

**Comment by yesoreyeram** [comment, marked as answer]
I know the root cause of pingdom api returning bad request. Pingdom API doesn't like the `Content-Type : application/json` header set by the infinity datasource. So setting a blank `Content-Type` header in the datasource config will help. Let us know if this helps.

You just need two headers 

`Authorization` : `Bearer YOUR_TOKEN`
`Content-Type` : ` `

![image](https://user-images.githubusercontent.com/153843/129335632-fe49c983-1f1a-410b-b978-0f312266fade.png)
![image](https://user-images.githubusercontent.com/153843/129335475-f3fd8363-bafe-4889-9084-ceaa89335576.png)

**Reply by smm0864** [reply]
thank you very much. that solved the problem 😃
